apps/form-gateway/src/gateway/router.ts

```diff
@@ -45,15 +47,52 @@ export function verifyCaptcha(logger: Logger, RECAPTCHA_SECRET: string, SCORE_TH
   };
 }
 
+async function getSimpleFormResponse(
+  formApiUrl: URL,
+  criteria: string,
+  tenantId: AdspId,
+  tokenProvider: TokenProvider
+): Promise<SimpleFormResponse> {
+  const token = await tokenProvider.getAccessToken();
+
+  const obj = JSON.parse(criteria);
+
+  //formId is contained in the last item of the split array
+  const formId = obj.recordIdContains ? obj.recordIdContains.split('/').at(-1) : '';
+
+  const targetPath = `${formApiUrl.toString()}form/v1/forms/${formId}`;
```

Comment:
Look at `DirectoryService.getResourceUrl()`. The directory object returned by the SDK can directly resolve a resource URL for you.